Specifications/Language/3_Expressions/ConditionalBranching.md

```diff
@@ -35,9 +32,8 @@ You can also express simple branching in the form of a [conditional expression](
 
 ## Target-specific restrictions
 
-The tight integration between control-flow constructs and quantum computations poses a challenge for current quantum hardware. Certain quantum processors do not support branching based on measurement outcomes. As such, comparison for values of type `Result` will always result in a compilation error for Q# programs that are targeted to run on such hardware. 
-
-Other quantum processors support specific kinds of branching based on measurement outcomes. The more general `if` statements supported in Q# are compiled into suitable instructions that can be run on such processors. The imposed restrictions are that values of type `Result` may only be compared as part of the condition within `if` statements in operations. Furthermore, the conditionally run blocks cannot contain any return statements or update mutable variables that are declared outside that block. 
+The tight integration between control-flow constructs and quantum computations poses a challenge for current quantum hardware. Certain quantum processors do not support branching based on measurement outcomes. As such, comparison for values of type `Result` will always result in a compilation error for Q# programs that are targeted to run on such hardware.
 
+Other quantum processors support specific kinds of branching based on measurement outcomes. The more general `if` expressions supported in Q# are compiled into suitable instructions that can be run on such processors. The imposed restrictions are that values of type `Result` may only be compared as part of the condition within `if` expressions in operations. Furthermore, the conditionally run blocks cannot contain any `return` expressions or update mutable variables that are declared outside that block.
```

Comment:
The language around not supporting return or mutable updates in blocks that are conditioned on measurement comparison is old language that is not true once the adaptive profile comes into play. I'm not sure if that is worth addressing now though.

Reply:
I'd rather not touch on that now, and we can re-evaluate the language once the adaptive profile is ready.